### DISCUSSION TITLE
MoodNft test for enums vs URI string

### DISCUSSION THREAD

**Discussion by jaimebarrancos** [Cyfrin/foundry-full-course-cu#282] [Q&A] [answered]
My test script using enums:
```solidity
function testFlipTokenToSad() public {
        vm.startPrank(USER);
        moodNft.mintNft();
        moodNft.flipMood(0);
        vm.stopPrank();

        assert(moodNft.tokenIdToMood(0) == MoodNft.Mood.SAD);
    }
```
Patrick's test script:
```solidity
function testFlipTokenToSad() public {
      vm.prank(USER);
      moodNft.mintNft();

      vm.prank(USER);
      moodNft.flipMood(0);

      assert(
          keccak256(abi.encodePacked(moodNft.tokenURI(0))) ==
              keccak256(abi.encodePacked(SAD_MOOD_URI))
      );
}
```
Why do one over the other? I think mine is simpler so what are the advantages of his script?

> Also, couldn't he just do ```assert(moodNft.tokenURI(0) == SAD_MOOD_URI)```

**Comment by cromewar** [comment, marked as answer]
Hello @jaimebarrancos 
Well all ways leads to Rome right? so there is not a unique way to do something. So your solution is actually good. 
Now, the token URI is more "unique" than the mood. If you have let's say a ERC1155 the token ids could match even tough the collection is no the same. But using the token uri which encoded is represented by a hash ensures you that you are going to actually have a unique identifier.

Anyways, as I said there is not a only way way to solve this and your solution is equally valid.
